qutip/core/tensor.py

```diff
@@ -467,4 +467,4 @@ def expand_operator(oper, dims, targets, dtype=None):
     for i, ind in enumerate(rest_pos):
         new_order[ind] = rest_qubits[i]
     id_list = [identity(dims[i]) for i in rest_pos]
-    return tensor([oper] + id_list).permute(new_order)
+    return tensor([oper] + id_list).permute(new_order).to(dtype)
```

Comment:
one question;  is there a reason the dtype defaults to CSR if its not specified or set in settings?  why not default to the dtype of `oper`?

Reply:
This is what we want to avoid.
If `oper` is dense, once extended, the output will be quite sparse:
```
oper = rand_herm(2, dtype="Dense")
expand_operator(oper, [2, 2, 2, 2, 2], [2])
```
The output will be a 32*32 matrix with 4 non-zeros elements...

Reply:
ah ok, that's a bit clearer, thanks!